libs/flux/build.go

```diff
@@ -211,11 +211,19 @@ func (l *Library) build(ctx context.Context, logger *zap.Logger) (string, error)
 	if targetString != "" {
 		cmd.Args = append(cmd.Args, "--target", targetString)
 
-		// Remove CC, CXX, and AR from the environment if the target is not ourselves.
-		// These variables interfere with the rust compiler toolchain's build.rs files.
-		cmd.Env = removeEnvVar(cmd.Env, "CC")
-		cmd.Env = removeEnvVar(cmd.Env, "CXX")
-		cmd.Env = removeEnvVar(cmd.Env, "AR")
+		// If the target environment is not also the host environment, CC set to "xcc" will
+		// be incorrect, and will interfere with build time compilation. For linux, unsetting the
+		// value is fine, as rust will correctly assume gcc. For macOS, however, we must explicitly
+		// set CC to clang.
+		if l.Target.OS == "darwin" {
```

Comment:
We require using `clang` in our docs anyway so can we just always set `CC` to `clang` on every platform when invoking `cargo`? I'd much prefer that over special casing a specific target os.

Reply:
Our cross compilers, for linux, are `gcc`, and so we'll get a similar error for linux if we try and force it to `clang`, though inverse (looking for the underscore symbol when it's not there).

Reply:
To be clear, this is because we use `cc` in the `build.rs` - we need to use `clang` for `CC` on darwin because `osxcross` expects the build time generated symbols to be compatible with `clang`, not `gcc`. We could unset it on linux, and the `cc` crate would just guess `gcc`, which is fine.

Truthfully, `clang` (and llvm in general) handles cross compilation better than `gcc`, which is why they tend to be incompatible. Unfortunately, our linux cross compilers are all gcc, so we can't set it to be gcc across the board.